### ISSUE TITLE
"Estimated fee out of range" intermittent failure in feature_fee_estimation.py

### ISSUE THREAD

**Issue by jonatack** [bitcoin/bitcoin#25179]
https://cirrus-ci.com/task/4707498732027904 on [nowallet libbitcoinkernel] [bionic]

```bash
 node1 2022-05-20T14:25:48.635128Z [httpworker.3] [policy/fees.cpp:365] [EstimateMedianVal] FeeEst: 1 > 95% decay 0.99520: feerate: 307809 from (301403 - 348912) 100.00% 34.4/(34.4 0 mem 0.0 out) Fail: (247965 - 301403) 86.60% 32.3/(32.3 5 mem 0.0 out) 
 node1 2022-05-20T14:25:48.635669Z [httpworker.3] [policy/fees.cpp:365] [EstimateMedianVal] FeeEst: 1 > 95% decay 0.99931: feerate: 732088 from (515502 - 1e+99) 100.00% 156.5/(156.5 0 mem 0.0 out) Fail: (247965 - 515502) 93.92% 154.6/(154.6 10 mem 0.0 out) 
 test  2022-05-20T14:25:48.638000Z TestFramework (ERROR): Assertion failed 
                                   Traceback (most recent call last):
                                     File "/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/test_framework/test_framework.py", line 133, in main
                                       self.run_test()
                                     File "/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/feature_fee_estimation.py", line 297, in run_test
                                       self.sanity_check_estimates_range()
                                     File "/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/feature_fee_estimation.py", line 205, in sanity_check_estimates_range
                                       check_estimates(self.nodes[1], self.fees_per_kb)
                                     File "/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/feature_fee_estimation.py", line 114, in check_estimates
                                       check_raw_estimates(node, fees_seen)
                                     File "/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/feature_fee_estimation.py", line 79, in check_raw_estimates
                                       f"Estimated fee ({feerate}) out of range ({min(fees_seen)},{max(fees_seen)})"
                                   AssertionError: Estimated fee (0.00732088) out of range (3.9370078740157485e-05,0.007271477272727272)
 test  2022-05-20T14:25:48.639000Z TestFramework (DEBUG): Closing down network thread 
 test  2022-05-20T14:25:48.690000Z TestFramework (INFO): Stopping nodes 
 test  2022-05-20T14:25:48.690000Z TestFramework.node0 (DEBUG): Stopping node 
```

I initially thought this might be the same as #23165 (and https://github.com/bitcoin/bitcoin/issues/11800 and https://github.com/bitcoin/bitcoin/issues/20725) that return *Estimated fee (a) larger than last fee (b) for lower number of confirms.*

But this one is *Estimated fee out of range*, so opening this issue and linking to the other ones.

**Comment by achow101** [comment]
https://cirrus-ci.com/task/5693607045758976

**Comment by aureleoules** [comment]
https://cirrus-ci.com/task/6270458164150272?logs=ci#L3318

**Comment by fanquake** [comment]
Should this have been closed with #26517?

**Comment by MarcoFalke** [comment]
Is this still an issue with a recent version of Bitcoin Core? If yes, what are the steps to reproduce?
